tests/vmi_configuration_test.go

```diff
@@ -1107,6 +1121,67 @@ var _ = Describe("Configurations", func() {
 		})
 	})
 
+	Context("with CPU request settings", func() {
+		defaultCPURequestKey := "cpu-request"
+
+		AfterEach(func() {
+			cfgMap, err := virtClient.CoreV1().ConfigMaps(namespaceKubevirt).Get(kubevirtConfig, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			cfgMap.Data[defaultCPURequestKey] = ""
+
+			_, err = virtClient.CoreV1().ConfigMaps(namespaceKubevirt).Update(cfgMap)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should set CPU request from VMI spec", func() {
+			vmi := tests.NewRandomVMI()
+			vmi.Spec.Domain.Resources.Requests[kubev1.ResourceCPU] = resource.MustParse("500m")
+			tests.RunVMIAndExpectLaunch(vmi, false, 30)
+
+			readyPod := tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
+			computeContainer := getComputeContainerOfPod(readyPod)
+			cpuRequest := computeContainer.Resources.Requests[kubev1.ResourceCPU]
+			Expect(cpuRequest.String()).To(Equal("500m"))
+		})
+
+		It("should set CPU request when it is not provided", func() {
+			vmi := tests.NewRandomVMI()
+			vmi.Spec.Domain.Resources = v1.ResourceRequirements{
+				Requests: kubev1.ResourceList{
+					kubev1.ResourceMemory: resource.MustParse("64M"),
+				},
+			}
+			tests.RunVMIAndExpectLaunch(vmi, false, 30)
+
+			readyPod := tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
+			computeContainer := getComputeContainerOfPod(readyPod)
+			cpuRequest := computeContainer.Resources.Requests[kubev1.ResourceCPU]
+			Expect(cpuRequest.String()).To(Equal("100m"))
+		})
+
+		It("should set CPU request from kubevirt-confg", func() {
+			cfgMap, err := virtClient.CoreV1().ConfigMaps(namespaceKubevirt).Get(kubevirtConfig, metav1.GetOptions{})
+			Expect(err).To(BeNil())
+			cfgMap.Data[defaultCPURequestKey] = "800m"
+			_, err = virtClient.CoreV1().ConfigMaps(namespaceKubevirt).Update(cfgMap)
```

Comment:
In theory there can be a race condition between detecting the config change and starting the VMI. In practice I don't think it matters for as long as the cluster is not in serious trouble, were we don't get reasonable results anyway then.

Reply:
Right, note that it is done like that also in "should set machine type from kubevirt-config" and the tests don't seem to be affected by this